### PR TITLE
upx: lzmasdk-9.38

### PR DESCRIPTION
**Pull request by denji**

### CODE REVIEW

File: Library/Formula/upx.rb [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-    ENV['UPX_LZMA_VERSION'] = '0x465'
+    inreplace "src/compress_lzma.cpp", "C/Types.h", "C/7zTypes.h"
+    (buildpath/"lzmasdk").install resource("lzma")
+    ENV["UPX_LZMADIR"] = buildpath/"lzmasdk"
+    ENV["UPX_LZMA_VERSION"] = "0x938"
     system "make all"
-    bin.install  'src/upx.out' => 'upx'
-    man1.install 'doc/upx.1'
+    bin.install  "src/upx.out" => "upx"
+    man1.install "doc/upx.1"
```
Comment (mikemcquaid) [inline Library/Formula/upx.rb]: Is there a `make install` we can use?

File: Library/Formula/upx.rb [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
   end
 end
```
Comment (mikemcquaid) [inline Library/Formula/upx.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/upx.rb (hunk 1 above)
Comment (denji) [inline Library/Formula/upx.rb]: make: *** No rule to make target `install'.  Stop.

Comment (mikemcquaid) [inline Library/Formula/upx.rb]: :+1:

Comment (denji) [inline Library/Formula/upx.rb]: ==> Finishing up
ln -s ../Cellar/upx/3.91_1/bin/upx upx
ln -s ../../../Cellar/upx/3.91_1/share/man/man1/upx.1 upx.1
==> Summary
🍺  /usr/local/Cellar/upx/3.91_1: 8 files, 1.8M, built in 7 seconds

**Comment by mikemcquaid** [comment]
install is failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/22797/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_upx/

**Comment by denji** [comment]
@mikemcquaid `--verbose`? I have no problems with building, 

```
~ brew install ~/Desktop/upx.rb --HEAD
==> Cloning https://www.pysol.org:4443/hg/upx.hg
Updating /Library/Caches/Homebrew/upx--hg
==> Downloading https://downloads.sourceforge.net/project/sevenzip/LZMA%20SDK/lzma938.7z
Already downloaded: /Library/Caches/Homebrew/upx--lzma-938.7z

7-Zip (A) [64] 9.20  Copyright (c) 1999-2010 Igor Pavlov  2010-11-18
p7zip Version 9.20 (locale=utf8,Utf16=on,HugeFiles=on,4 CPUs)

Processing archive: /Library/Caches/Homebrew/upx--lzma-938.7z

Extracting  C/Util/LzmaLib/makefile
....

Everything is Ok

Files: 623
Size:       4176841
Compressed: 896456
==> make all
🍺  /usr/local/Cellar/upx/HEAD: 8 files, 1.7M, built in 11 seconds
```

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.10.3
BuildVersion:	14D98g
xcode-select version 2339
```

**Comment by mikemcquaid** [comment]
> @mikemcquaid --verbose? No problems with building,

Regardless, we can't include it until this builds on the bot I'm afraid.

**Comment by jacknagel** [comment]
Makefile:

```make
#
# UPX toplevel Makefile - needs GNU make 3.81 or better
#
# Copyright (C) 1996-2013 Markus Franz Xaver Johannes Oberhumer
#

MAKEFLAGS += -rR
.SUFFIXES:
export SHELL = /bin/sh

ifneq ($(findstring $(firstword $(MAKE_VERSION)),3.77 3.78 3.78.1 3.79 3.79.1 3.80),)
$(error GNU make 3.81 or better is required)
endif

srcdir = .
top_srcdir = .
include $(wildcard $(top_srcdir)/Makevars.global ./Makevars.local)


# info: src/stub needs special build tools
BUILD_STUB = 0
ifneq ($(wildcard $(HOME)/local/bin/bin-upx/.),)
BUILD_STUB = 1
endif
ifneq ($(wildcard $(HOME)/bin/bin-upx/.),)
BUILD_STUB = 1
endif


default:
	@echo "UPX info: please choose a target for 'make'"

all mostlyclean clean distclean maintainer-clean:
ifeq ($(BUILD_STUB),1)
	$(MAKE) -C src/stub $@
endif
	$(MAKE) -C src $@
	$(MAKE) -C doc $@


.PHONY: default all mostlyclean clean distclean maintainer-clean

```

Somehow BUILD_STUB is getting set to 1?

**Comment by jacknagel** [comment]
@BrewTestBot test this please

**Comment by jacknagel** [comment]
huh, worked. shrug

**Pull request closed without merging**